pyuvdata/uvdata/tests/test_mir_meta_data.py

```diff
@@ -0,0 +1,932 @@
+def mir_sp_data(mir_data_main):
+    yield mir_data_main.sp_data.copy()
+
+
+@pytest.fixture(scope="function")
+def mir_eng_data(mir_data_main):
+    yield mir_data_main.eng_data.copy()
+
+
+@pytest.fixture(scope="function")
+def mir_codes_data(mir_data_main):
+    yield mir_data_main.codes_data.copy()
+
+
+def test_mir_meta_init(mir_data):
+    """
+    Test that the initialization of MirMetaData objects behave as expected. This
+    includes subtypes that are part of the MirParser class.
+    """
+    attr_list = list(mir_data._metadata_attrs.keys())
+    # Drop "ac_data", since it's a synthetic table
+    attr_list.remove("ac_data")
+
+    for item in attr_list:
+        # antpos_data is a little special, since it's a text file, so we can't use the
+        # generic read function for MirMetaData here.
+        attr = getattr(mir_data, item)
+        if item != "antpos_data":
+            # Read in the metadata w/ the generic read
+            meta_attr = MirMetaData(
+                attr._filetype,
+                attr.dtype,
+                attr._header_key,
+                attr._binary_dtype,
+                attr._pseudo_header_key,
+                mir_data.filepath,
+            )
+            # Now make a placeholder subtype
+            new_attr = type(attr)()
+
+            # Plug in the __dict__, which _should_ transfer over all the writable attrs
+            new_attr.__dict__ = meta_attr.__dict__
+            assert attr == new_attr
+
+        new_attr = type(attr)(mir_data.filepath)
+        assert attr == new_attr
+
+
+def test_mir_meta_iter(mir_data):
+    """
+    Test that MirMetaData objects iterate as expected, which is that they should yield
+    the full group for each index position (similar to ndarray).
+    """
+    # Test the unflagged case
+    in_data = mir_data.in_data
+    for idx, item in enumerate(in_data):
+        assert in_data._data[idx] == item
+
+    # Now the flagged case
+    in_data._mask[1::2] = False
+    for idx, item in enumerate(in_data):
+        assert in_data._data[2 * idx] == item
+
+
+def test_mir_meta_copy(mir_in_data):
+    """
+    Verify that the copy operation of MirMetaData produces a duplicate dataset that is
+    NOT a reference to the same data in memory.
+    """
+    other = mir_in_data.copy()
+    for item in vars(other):
+        this_attr = getattr(mir_in_data, item)
+        other_attr = getattr(other, item)
+
+        # Nones str, and tuple can be duplicates, since they're both immutable.
+        if not (isinstance(this_attr, (str, tuple)) or this_attr is None):
+            assert this_attr is not other_attr
+
+        assert np.all(this_attr == other_attr)
+
+
+@pytest.mark.parametrize(
+    "comp_obj,err_msg", [[0, "Cannot compare MirInData with int."]]
+)
+def test_mir_meta_eq_errs(mir_in_data, comp_obj, err_msg):
+    with pytest.raises(ValueError) as err:
+        mir_in_data.__eq__(comp_obj)
+    assert str(err.value).startswith(err_msg)
```

Comment:
```suggestion
    with pytest.raises(ValueError, match=err_msg):
        mir_in_data.__eq__(comp_obj)
```

Can do similar things throughout.

Reply:
Cleaned up in e1161632.